### DISCUSSION TITLE
ScriptReference/HandleUtility.FindNearestVertex

### DISCUSSION THREAD

**Discussion by giscus** [Developer-Notes-Extension/Unity-Notes#599] [Notes]
# ScriptReference/HandleUtility.FindNearestVertex

https://docs.unity3d.com/ScriptReference/HandleUtility.FindNearestVertex.html

**Comment by FreyaHolmer** [comment]
Be **very careful** using the overload without the `objectsToIgnore` parmeter - Unity will populate this on its own depending on editor state, so I recommend always using the overload with both `objectsToSearch` and `objectsToIgnore` if you want full control over picking.

In my case, I ran into hours of debugging because when using a `PositionHandle`, holding V for vertex snapping, unity's code path decides what gets ignored or not in this seemingly unrelated function.
